hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -349,7 +349,7 @@ public class HoodieWriteConfig extends HoodieConfig {
 
   public static final ConfigProperty<String> EMBEDDED_TIMELINE_SERVER_USE_ASYNC_ENABLE = ConfigProperty
       .key("hoodie.embed.timeline.server.async")
-      .defaultValue("false")
+      .defaultValue("true")
       .withDocumentation("Controls whether or not, the requests to the timeline server are processed in asynchronous fashion, "
```

Comment:
What is exactly the effect of this param, i'm scared whether it can cause data loss like https://github.com/apache/hudi/pull/6179

Reply:
I see. The intent is to improve the throughput by requesting the timeline server asynchronously. Internally, we have run a long-running test (30+ commits) with this config but did not see any data loss. Our validations were based on count. However, if you have any concern let me know. We can keep it false by default for now. Data loss issues are more critical.

Reply:
30+ commits is too few to reproduce the problem, in #6179 , we run about 2000+ commits to reproduce the problem.